### PR TITLE
Replace query queues with resource groups

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

**Comment by raghavsethi** [comment]
@electrum Can we land this?

**Comment by wenleix** [comment]
@electrum : +1 on can you land this ? :)

**Comment by electrum** [comment]
@wenleix waiting on review (hint hint) and @raghavsethi to fix `TestQueryStateInfoResource` (it breaks when resource groups are enabled)

File: presto-main/src/main/java/com/facebook/presto/server/CoordinatorModule.java [hunk 1]
```diff
@@ -175,14 +171,8 @@ protected void setup(Binder binder)
...
         binder.bind(ResourceGroupManager.class).to(InternalResourceGroupManager.class);
-        binder.bind(LegacyResourceGroupConfigurationManagerFactory.class).in(Scopes.SINGLETON);
-        if (buildConfigObject(FeaturesConfig.class).isResourceGroupsEnabled()) {
-            binder.bind(QueryQueueManager.class).to(InternalResourceGroupManager.class);
-        }
-        else {
-            binder.bind(QueryQueueManager.class).to(SqlQueryQueueManager.class).in(Scopes.SINGLETON);
-            binder.bind(new TypeLiteral<List<QueryQueueRule>>() {}).toProvider(QueryQueueRuleFactory.class).in(Scopes.SINGLETON);
-        }
+        binder.bind(LegacyResourceGroupConfigurationManager.class).in(Scopes.SINGLETON);
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/server/CoordinatorModule.java]: Why do we need a legacy manager at all?

Comment (electrum) [reply presto-main/src/main/java/com/facebook/presto/server/CoordinatorModule.java]: It is used by `InternalResourceGroupManager` when there is no resource group configuration. I agree it should be removed, but that's a separate project (I didn't see an easy way).

**Comment by wenleix** [comment]
Random thought: Should we just even deprecate `QueryQueueManager` and assume there is only `ResourceGroupManager` ?

**Comment by electrum** [comment]
@wenleix Sounds fine to me, feel free to send a PR

**Pull request merged**